### PR TITLE
fix: warn about import assertions when using typescript

### PR DESCRIPTION
**Pull request by dsherret**
1. On emit, checks for the prescence of import assertions.
1. Warns and doesn't store the parsed source in the emit cache in this case.

### CODE REVIEW

File: cli/emit.rs [hunk 1]
```diff
@@ -276,11 +289,80 @@ impl<'a> EmitParsedSourceHelper<'a> {
...
+  let Some(module) = parsed_source.program_ref().as_module() else {
+    return false;
+  };
+
+  let mut had_import_assertion = false;
+  for item in &module.body {
+    match item {
+      deno_ast::swc::ast::ModuleItem::ModuleDecl(decl) => match decl {
+        deno_ast::swc::ast::ModuleDecl::Import(n) => {
+          if n.with.is_some()
```
Comment (bartlomieju) [inline cli/emit.rs:334]: Can you also handle `import("./data.json", { assert: { type: "json" }})`?

Comment (dsherret) [reply cli/emit.rs:334]: Oh, that's annoying. I'll update it.

Comment (dsherret) [reply cli/emit.rs:334]: We don't need to do it for that because v8 will warn about it.

**Review by bartlomieju — APPROVED** [review]

**Review by littledivy — APPROVED** [review]
lgtm, cool fix. I guess swc should have an option to preserve import attr syntax?

**Comment by dsherret** [comment]
> I guess swc should have an option to preserve import attr syntax?

It's probably not worth the trouble.

**Pull request merged**